packages/collector/test/tracing/messaging/node-rdkafka/producer.js

```diff
@@ -0,0 +1,200 @@
+/*
+ * (c) Copyright IBM Corp. 2022
+ */
+
+'use strict';
+
+require('../../../../')({
+  tracing: {
+    kafka: {
+      headerFormat: 'string'
+    }
+  }
+});
+const Kafka = require('node-rdkafka');
+const fetch = require('node-fetch');
+const agentPort = process.env.INSTANA_AGENT_PORT || 42699;
+const logPrefix = `rdkafka Producer (${process.pid}):\t`;
+const log = require('@instana/core/test/test_util/log').getLogger(logPrefix);
+const express = require('express');
+const port = process.env.APP_PORT || 3216;
+const enableDeliveryCb = process.env.RDKAFKA_PRODUCER_DELIVERY_CB === 'true';
+
+const app = express();
+
+const topic = 'rdkafka-topic';
+
+function getProducer(isStream = false) {
+  /** @type {Kafka.Producer | Kafka.ProducerStream} */
+  let _producer;
+  const producerOptions = {
+    'client.id': 'rdkafka-producer',
+    'metadata.broker.list': '127.0.0.1:9092',
+    // Enable to receive message payload in "delivery reports" event. This is the only way to know when a message
+    // was successfully sent. But we cannot rely on it, nor set this option without the customer's knowledge.
+    dr_cb: enableDeliveryCb
+  };
+
+  if (isStream) {
+    _producer = Kafka.Producer.createWriteStream(
+      producerOptions,
+      {},
+      {
+        topic: topic,
+        // This option is needed in order to use headers as Producer stream.
+        // This information needs to be very clear in the documentation.
+        // In the instrumentation we can check Buffer.isBuffer, like it's done here:
+        // https://github.com/Blizzard/node-rdkafka/blob/master/lib/producer-stream.js#L232
+        // Without this option, the stream expects only the message as a Buffer.
+        objectMode: true
```

Comment:
Should we also have a minimal test case where objectMode is false?

Of course, there would be no trace correlation but we can at least verify that we do not break anything when the option is set to false. The expectations for that test case would be that and exit and an entry are produced, but they would not be part of the same trace (different trace IDs).

Reply:
Will take over too

Reply:
good catch. I missed that one